server/server.js

```diff
@@ -0,0 +1,93 @@
+var express = require('express');
+var app = express();
+var port = 3000;
+var cors = require('cors')
+var task = require("./task");
+app.use(cors());
+
+
+//express
+var bodyParser = require("body-parser");
+app.use(bodyParser.json());
+
+app.post("/todo", function(req, res, next) {
+  task.create({
+    title: req.body.title,
+    dueDate: req.body.dueDate,
+    address: req.body.address,
+    lat: req.body.lat,
+    lng: req.body.lng
+  }, function(err, task) {
+    if (err) {
+      res.status(500);
+      res.send(err);
+    } else {
+      res.status(201);
+      res.send(task);
+    }
+  });
+})
+app.delete("/todo/:id", function(req, res, next) {
+  var id = '' + req.params.id;
+  task.remove({
+    _id: id
+  }, function(err, result) {
+    if (err) {
+      res.status(500);
+      next("Internal server error.");
+    } else if (result.n == 0) {
+      res.status(404); // Not found
+      next("Not found.");
+    } else {
+      res.status(200);
+      res.send(result);
+    }
+  });
+})
+app.put("/todo/:id", function(req, res, next) {
+  var id = '' + req.params.id;
+  task.update({
+    _id: '' + id
+  }, {
+    title: req.body.title,
+    dueDate: req.body.dueDate,
+    address: req.body.address,
+    lat: req.body.lat,
+    lng: req.body.lng
+  }, function(err, result) {
+    if (err) {
+      res.status(500);
+      next("Internal server error.");
+    } else if (result.n == 0) {
+      res.status(404); // Not found
+      next("Not found.");
+    } else {
+      res.status(200);
+      res.send(result);
+    }
+  });
+})
+
+app.get("/todo", function(req, res) {
+  var stream = task.find().stream();
+  if (stream) {
+    var results = [];
+    stream.on('data', function(doc) {
+      delete doc.__v;
+      results.push(doc);
+    }).on("error", function(err) {
+      res.status(500);
+      res.send(err);
```

Comment:
Oh, should we use next() here instead?

Reply:
You think that's better? I just didn't understand what next is used for? Anyways, suggestion accepted.

Reply:
You think that's better? I just didn't understand what next is used for? Anyways, suggestion accepted.

Reply:
I don't totally understand it either, but I think it keeps the server from hanging on errored out requests by instructing it to go on to the next thing. [This page](https://expressjs.com/en/guide/error-handling.html) says it's important for garbage collection:
> Notice that when not calling “next” in an error-handling function, you are responsible for writing (and ending) the response. Otherwise those requests will “hang” and will not be eligible for garbage collection.
So maybe we should use it in addition to the res.send()? Or maybe it's not necessary since we are completing the request? I'm not sure 😕